### ISSUE TITLE
NVIDIA Jetson Support

### ISSUE THREAD

**Issue by maverich** [KevinOConnor/klipper#3238]
Did anyone tested klipper on NVIDIA Jetson? asking to use on board CUDA processors which is not provided by any other boards

**Comment by klipper-gitissuebot** [comment]
Hi @maverich,

It did not look like there was a Klipper log file attached to this ticket. The log file has been engineered to answer common questions the Klipper developers have about the software and its environment (software version, hardware type, configuration, event timing, and hundreds of other questions).

Unfortunately, too many people have opened tickets without providing the log. That consumes developer time; time that would be better spent enhancing the software. If this ticket references an event that has occurred while running the software then the Klipper log must be attached to this ticket. Otherwise, this ticket will be automatically closed in a few days.

For information on obtaining the Klipper log file see: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md

The log can still be attached to this ticket - just add a comment and attach the log to that comment.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by danielfmo** [comment]
To what gain? 
I can understand the curiosity and discovery of this but the truth is that an raspberry pi0 alone is powerful enough to cover the majority of uses cases (klipper wise alone)

**Comment by The-Monkey-King** [comment]
Currently, I do not have one, but reading the specs, it does work with Linux. So in theory, it should work. 
However, the use cases for Jeston (aka Jetson NANO) would best suit it for virtual printing - as in running best case/testing printing simulations and not real printing. There is only so much bandwidth, including physical limiters (speed, heat, cooling, extrusion, plasticity of material), that current printing techniques can use. 

Short answer: Yes (Linux enabled) but YMMV
Use case: ?

**Comment by maverich** [comment]
I guess there is some missing information about use cases, I need the CUDA power mostly for running simple deep learning models. What these deep learning models can do? Imagine a computer vision model running with camera feed detecting faults. So you dont have to stream the footage to a backend service anymore. Or a model trying to increase the speed as much as possible until it sees some side effects? when it is just simple machine learning, it works in normal raspberry pi easily but when it comes to computer vision or long sequences, then you need the power of a GPU

**Comment by gui7vaz** [comment]
This is something I always imagined for 3D printers.

I work for automotive companies and we use this kind of feature to inspect
glue application in real time. Not with ML, just some customized vision
features.
(example of a specific devices we use)
https://www.quiss.net/products/rt-vision-t/
https://www.isravision.com/en/ready-to-use/bead-inspection/beadmaster3d/

**Comment by KevinOConnor** [comment]
I don't know of anyone using that particular machine, but lots of people run the Klipper host code on Linux machines other than a Raspberry PI.  https://www.klipper3d.org/FAQ.html#can-i-run-klipper-on-something-other-than-a-raspberry-pi-3

-Kevin

**Comment by The-Monkey-King** [comment]
Oh, ok. On this topic -- I got to see SAS (the analytics company) showcase their visual analytics programs using just a Raspberry Pi. Basically, it was a bottling line that would visually detect when the soda bottle did not match a model (ie. cap off, bottle on the side, bottle wrong color, etc). This was the Neuro [event](https://nips.cc/Expo/Conferences/2018/Schedule?demo_id=17).

Here is abstract [one](https://www.sas.com/content/dam/SAS/support/en/sas-global-forum-proceedings/2020/4432-2020.pdf) and abstract [two](https://www.sas.com/content/dam/SAS/support/en/sas-global-forum-proceedings/2020/4432-2020.pdf).

@KevinOConnor - I've been able to install and run Klipper in an Oracle VM (VirtualBox) of Debian on the Windows 10 platform. 
On my to-do list:
 1) Install in VM on Catalina MacOS (same rig as above - Ryzentosh)
 2) Install directly on Win10 platform, using Python3/C and see if I can have it run a Win10 service
 3) Same as #2 on MacOS platform

Why? Uhm,  because it's there.. and basically slimline my computing environment. Also, thinking of networking my 3D printers (USB-to-Ethernet), like Prusa does at his plant. 
 
> 
> I guess there is some missing information about use cases, I need the CUDA power mostly for running simple deep learning models. What these deep learning models can do? Imagine a computer vision model running with camera feed detecting faults. So you dont have to stream the footage to a backend service anymore. Or a model trying to increase the speed as much as possible until it sees some side effects? when it is just simple machine learning, it works in normal raspberry pi easily but when it comes to computer vision or long sequences, then you need the power of a GPU
